packages/records/lib/models/records.ts

```diff
@@ -0,0 +1,417 @@
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc.js';
+import { db } from '../db/client.js';
+import type {
+    FormattedRecord,
+    FormattedRecordWithMetadata,
+    GetRecordsResponse,
+    LastAction,
+    ReturnedRecord,
+    UnencryptedRecord,
+    UpsertSummary
+} from '../types.js';
+import { decryptRecordData, decryptRecords, encryptDataRecords } from '../utils/encryption.js';
+import { RECORDS_TABLE } from '../constants.js';
+import { removeDuplicateKey, getUniqueId } from '../helpers/uniqueKey.js';
+import { logger } from '../utils/logger.js';
+import { resultErr, resultOk } from '@nangohq/shared';
+import type { Result } from '@nangohq/shared';
+
+dayjs.extend(utc);
+
+const BATCH_SIZE = 1000;
+
+export async function getRecords(
```

Comment:
Agree with feedback that @bodinsamuel made in another PR that maybe an object is easier to reason about and manage esp on internal only calls like this.

Reply:
I don't remember @bodinsamuel comment. Can you refresh my memory? :D

Reply:
Sorry, it wasn't made in this PR, but rather here: https://github.com/NangoHQ/nango/pull/1903#discussion_r1541202375

Reply:
ah got it you mean for the parameters. Yes that make sense

Reply:
can be done in a subsequent PR if that brings too much change imo

Reply:
done